lib/vfs/src/host_fs.rs

```diff
@@ -0,0 +1,648 @@
+        libc::EINVAL => Err(FsError::InvalidInput),
+        _ => Err(FsError::IOError),
+    }
+}
+
+#[cfg(not(unix))]
+fn host_file_bytes_available(_raw_fd: i32) -> Result<usize, FsError> {
+    unimplemented!("host_file_bytes_available not yet implemented for non-Unix-like targets.  This probably means the program tried to use wasi::poll_oneoff")
+}
+
+/// A wrapper type around Stdout that implements `VirtualFile` and
+/// `Serialize` + `Deserialize`.
+#[derive(Debug, Default)]
+#[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
+pub struct Stdout;
+impl Read for Stdout {
+    fn read(&mut self, _buf: &mut [u8]) -> io::Result<usize> {
+        Err(io::Error::new(
+            io::ErrorKind::Other,
+            "can not read from stdout",
+        ))
+    }
+    fn read_to_end(&mut self, _buf: &mut Vec<u8>) -> io::Result<usize> {
+        Err(io::Error::new(
+            io::ErrorKind::Other,
+            "can not read from stdout",
+        ))
+    }
+    fn read_to_string(&mut self, _buf: &mut String) -> io::Result<usize> {
+        Err(io::Error::new(
+            io::ErrorKind::Other,
+            "can not read from stdout",
+        ))
+    }
+    fn read_exact(&mut self, _buf: &mut [u8]) -> io::Result<()> {
+        Err(io::Error::new(
+            io::ErrorKind::Other,
+            "can not read from stdout",
+        ))
+    }
+}
+impl Seek for Stdout {
+    fn seek(&mut self, _pos: io::SeekFrom) -> io::Result<u64> {
+        Err(io::Error::new(io::ErrorKind::Other, "can not seek stdout"))
+    }
+}
+impl Write for Stdout {
+    fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+        io::stdout().write(buf)
+    }
+    fn flush(&mut self) -> io::Result<()> {
+        io::stdout().flush()
+    }
+    fn write_all(&mut self, buf: &[u8]) -> io::Result<()> {
+        io::stdout().write_all(buf)
+    }
+    fn write_fmt(&mut self, fmt: ::std::fmt::Arguments) -> io::Result<()> {
+        io::stdout().write_fmt(fmt)
+    }
+}
+
+#[cfg_attr(feature = "enable-serde", typetag::serde)]
+impl VirtualFile for Stdout {
+    fn last_accessed(&self) -> u64 {
+        0
+    }
+    fn last_modified(&self) -> u64 {
+        0
+    }
+    fn created_time(&self) -> u64 {
+        0
+    }
+    fn size(&self) -> u64 {
+        0
+    }
+    fn set_len(&mut self, _new_size: u64) -> Result<(), FsError> {
+        debug!("Calling VirtualFile::set_len on stdout; this is probably a bug");
+        Err(FsError::PermissionDenied)
+    }
+    fn unlink(&mut self) -> Result<(), FsError> {
+        Ok(())
+    }
+    fn bytes_available(&self) -> Result<usize, FsError> {
+        // unwrap is safe because of get_raw_fd implementation
+        let host_fd = self.get_raw_fd().unwrap();
+
+        host_file_bytes_available(host_fd)
+    }
+    #[cfg(unix)]
+    fn get_raw_fd(&self) -> Option<i32> {
+        use std::os::unix::io::AsRawFd;
+        Some(io::stdout().as_raw_fd())
+    }
+    #[cfg(not(unix))]
```

Comment:
```suggestion
    #[cfg(windows)]
```

Reply:
See https://github.com/wasmerio/wasmer/pull/2491#discussion_r689570334. tl;dr it's fixed by https://github.com/wasmerio/wasmer/pull/2524.